app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -6,13 +6,20 @@ module SamlIdpAuthConcern
     # rubocop:disable Rails/LexicallyScopedActionFilter
     before_action :validate_saml_request, only: :auth
     before_action :validate_service_provider_and_authn_context, only: :auth
+    before_action :sign_out_if_forceauthn_is_true_and_user_is_signed_in, only: :auth
     before_action :store_saml_request, only: :auth
     before_action :check_sp_active, only: :auth
     # rubocop:enable Rails/LexicallyScopedActionFilter
   end
 
   private
 
+  def sign_out_if_forceauthn_is_true_and_user_is_signed_in
```

Comment:
This is mostly copied from the OIDC implementation of `prompt=login`, minus checking for SP handoff bounces (which I wasn't sure applied here).